homeassistant/components/homekit/type_air_purifiers.py

```diff
@@ -0,0 +1,452 @@
+            self.entity_id,
+            self.linked_filter_change_indicator_binary_sensor,
+            current_change_indicator,
+        )
+        self.char_filter_change_indication.set_value(current_change_indicator)
+
+    @callback
+    def _async_update_filter_life_level_event(
+        self, event: Event[EventStateChangedData]
+    ) -> None:
+        """Handle state change event listener callback."""
+        self._async_update_filter_life_level(event.data.get("new_state"))
+
+    @callback
+    def _async_update_filter_life_level(self, new_state: State | None) -> None:
+        """Handle linked filter life level sensor state change to update HomeKit value."""
+        if new_state is None:
+            return
+
+        current_life_level = float(new_state.state)
+        if (
+            self.char_filter_life_level
+            and self.char_filter_life_level.value != current_life_level
+        ):
+            _LOGGER.debug(
+                "%s: Linked filter life level sensor %s changed to %d",
+                self.entity_id,
+                self.linked_filter_life_level_sensor,
+                current_life_level,
+            )
+            self.char_filter_life_level.set_value(current_life_level)
+
+        if self.linked_filter_change_indicator_binary_sensor:
+            # Handled by its own event listener
+            return
+
+        current_change_indicator = (
+            FILTER_CHANGE_FILTER
+            if (current_life_level < THRESHOLD_FILTER_CHANGE_NEEDED)
+            else FILTER_OK
+        )
+        if (
+            not self.char_filter_change_indication
+            or self.char_filter_change_indication.value == current_change_indicator
+        ):
+            return
+
+        _LOGGER.debug(
+            "%s: Linked filter life level sensor %s changed to %d",
+            self.entity_id,
+            self.linked_filter_life_level_sensor,
+            current_change_indicator,
+        )
+        self.char_filter_change_indication.set_value(current_change_indicator)
+
+    @callback
+    def async_update_state(self, new_state: State) -> None:
+        """Update fan after state change."""
+        super().async_update_state(new_state)
+        # Handle State
+        state = new_state.state
+
+        if self.char_current_air_purifier_state is not None:
+            self.char_current_air_purifier_state.set_value(
+                CURRENT_STATE_PURIFYING_AIR
+                if state == STATE_ON
+                else CURRENT_STATE_INACTIVE
+            )
+
+        # Automatic mode is represented in HASS by a preset called Auto or auto
+        attributes = new_state.attributes
+        current_preset_mode = attributes.get(ATTR_PRESET_MODE)
+        if current_preset_mode is not None:
+            self.char_target_air_purifier_state.set_value(
+                TARGET_STATE_AUTO
+                if current_preset_mode.lower() == "auto"
```

Comment:
We should have coverage that `Auto` and `auto` both work.

Reply:
Done, and also found an edge case when working on this when there's only a single preset mode (it would crash in the `type_fans.py` constructor), so added a separate test for that, too.